### PR TITLE
CLN: follow-ups

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by jreback** [comment]
whoo hoo issue 40000!

**Comment by jbrockmendel** [comment]
looks like the test thats failing here is failing everywhere

**Comment by jreback** [comment]
> pandas/_libs/lib.html

got commited :_>

**Comment by jreback** [comment]
> looks like the test thats failing here is failing everywhere

can you xfail / fix ?

**Comment by jbrockmendel** [comment]
> got commited :_>

woops, will revert

> can you xfail / fix ?

was planning to try to fix it in a bit

**Comment by jreback** [comment]
> was planning to try to fix it in a bit

sure

File: pandas/core/array_algos/take.py [hunk 1]
```diff
@@ -57,20 +57,29 @@ def take_nd(
...
     if not isinstance(arr, np.ndarray):
         # i.e. ExtensionArray,
         # includes for EA to catch DatetimeArray, TimedeltaArray
         return arr.take(indexer, fill_value=fill_value, allow_fill=allow_fill)
 
     arr = np.asarray(arr)
+    return _take_nd_ndarray(arr, indexer, axis, out, fill_value, allow_fill)
+
+
+def _take_nd_ndarray(
```
Comment (jreback) [inline pandas/core/array_algos/take.py:75]: this is purely for typing right?

Comment (jbrockmendel) [reply pandas/core/array_algos/take.py:75]: yah mypy complained without it

Comment (jreback) [reply pandas/core/array_algos/take.py:75]: you can't just cast here?

Comment (jbrockmendel) [reply pandas/core/array_algos/take.py:75]: simon explained a couple days ago in a different thread that it involves ArrayLike being a TypeVar and not a union

Comment (jreback) [reply pandas/core/array_algos/take.py:75]: ok then

**Review by jreback — APPROVED** [review]

**Pull request merged**